### DISCUSSION TITLE
Omit a controller or controller function from auth

### DISCUSSION THREAD

**Discussion by ArslanAmeer** [brocoders/nestjs-boilerplate#1719] [General]
want to create a controller where all APIs or specific endpoints are accessible to all users. I couldn't find anything related to this in the documentation, so I need some help. For example, I have created a resource using the CLI and added properties to it. Now, I want to make a few functions accessible without requiring authorization.

How can I achieve this?

**Comment by Shchepotin** [comment]
@ArslanAmeer  You can remove `@UseGuards(AuthGuard('jwt'))` from the controller

**Reply by ArslanAmeer** [reply]
What if I want just one endpoint or controller function to work without authentication?
Otherwise, I can handle it by applying an Auth Guard to all the functions that require authentication and leaving it off for those that don't.

**Reply by Shchepotin** [reply]
> I can handle it by applying an Auth Guard to all the functions that require authentication and leaving it off for those that don't.

This way is correct 👍
